lib/engine/game/g_18_norway/steps/dividend.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/dividend'
+require_relative '../../../step/half_pay'
+require_relative '../../../step/minor_half_pay'
+
+module Engine
+  module Game
+    module G18Norway
+      module Step
+        class Dividend < Engine::Step::Dividend
+          DIVIDEND_TYPES = %i[payout half withhold].freeze
+          include Engine::Step::HalfPay
+
+          def share_price_change(entity, revenue = 0)
+            return {} if entity.minor?
+
+            price = entity.share_price.price
+            return { share_direction: :left, share_times: 1 } if revenue < price / 2
```

Comment:
You need to be careful with division: Opal translates the code to Javascript which only has floating point numbers. It's unlikely that this would be a problem (I assume the revenue will be a multiple of 10, so you're not going to be dealing with fractions), but multiplication is a bit safer than division.
```suggestion
            return { share_direction: :left, share_times: 1 } if revenue * 2 < price
```